README.md

```diff
@@ -291,43 +291,47 @@ As you can see, some of the properties — such as `metadata` — are loaded on
 - If you're only fetching one NFT — e.g. by using `findNftByMint` — then these properties will already be loaded.
 - If you're fetching multiple NFTs — e.g. by using `findNftsByMintLint` — then these properties will not be loaded and you will need to load them as and when you need them.
 
-In order to load these properties, you may use the `metadataLoader` and `masterEditionLoader` properties of the `Nft` object.
+In order to load these properties, you may run the `metadataTask` and `masterEditionTask` properties of the `Nft` object.
 
 ```ts
-await nft.metadataLoader.load();
-await nft.masterEditionLoader.load();
+await nft.metadataTask.run();
+await nft.masterEditionTask.run();
 ```
 
-After these two promises resolve, you should have access to the `metadata`, `masterEditionAccount` and `masterEdition` properties. Note that if a loader fails to load the data, an error will be thrown. You may change that behaviour by providing the `failSilently` option to the `load` method.
+After these two promises resolve, you should have access to the `metadata`, `masterEditionAccount` and `masterEdition` properties. Note that if a task fails to load the data, an error will be thrown.
 
-```ts
-await nft.metadataLoader.load({ failSilently: true });
-```
-
-Also, note that both `metadataLoader` and `masterEditionLoader` are instances of the `Loader` class which contains a bunch of helper methods. Here's an overview of the methods available in the `Loader` class:
+Also, note that both `metadataTask` and `masterEditionTask` are of type `Task` which contains a bunch of helper methods. Here's an overview of the methods available in the `Task` class:
 
 ```ts
-class Loader<T> {
-    public getStatus(): LoaderStatus;
-    public getResult(): T | undefined;
-    public getError(): unknown;
-    public isPending(): boolean;
-    public isLoading(): boolean;
-    public isLoaded(): boolean;
-    public isSuccessful(): boolean;
-    public isFailed(): boolean;
-    public isCanceled(): boolean;
+export type Task<T> = {
+    getStatus: () => TaskStatus;
+    getResult: () => T | undefined;
+    getError: () => unknown;
```

Comment:
Should this return `Error|string` possibly instead of something totally `unknown`?

Reply:
I'd love to make that more strict but unfortunately, with JavaScript, people can throw anything they want. In fact, if you try to type-hint a try/catch error, it will tell you it can only be type-hinted with `unknown` or `any`. Whilst, we'll eventually have control over our own `Error` type, we don't control what end-users will be throwing within the task callback which will bubble up into that `getError` method. Therefore, we don't have control over that error type and, sadly, it has to stay `unknown` to account for all possibilities. We could add another generic type with a sensible default like `Task<T, E = Error | string>` but I thought it was a bit overkill. Maybe we can leave it `unknown` for now and revisit later as we implement better error handling and as we get more feedback?